src/programs/raster/mdimtranslate.rs

```diff
@@ -0,0 +1,266 @@
+use crate::{
+    errors::*,
+    utils::{_last_null_pointer_err, _path_to_c_string},
+    Dataset,
+};
+use gdal_sys::{GDALMultiDimTranslate, GDALMultiDimTranslateOptions};
+use libc::{c_char, c_int};
+use std::{
+    borrow::Borrow,
+    cell::Cell,
+    ffi::CString,
+    mem::ManuallyDrop,
+    path::Path,
+    ptr::{null, null_mut},
+};
+
+/// Wraps a [GDALMultiDimTranslateOptions] object.
+///
+/// [GDALMultiDimTranslateOptions]: https://gdal.org/api/gdal_utils.html#_CPPv428GDALMultiDimTranslateOptions
+///
+pub struct MultiDimTranslateOption {
+    c_options: *mut GDALMultiDimTranslateOptions,
+}
+
+impl MultiDimTranslateOption {
+    /// See [GDALMultiDimTranslateOptionsNew].
+    ///
+    /// [GDALMultiDimTranslateOptionsNew]: https://gdal.org/api/gdal_utils.html#_CPPv431GDALMultiDimTranslateOptionsNewPPcP37GDALMultiDimTranslateOptionsForBinary
+    ///
+    pub fn new<S: Into<Vec<u8>>, I: IntoIterator<Item = S>>(args: I) -> Result<Self> {
+        // Convert args to CStrings to add terminating null bytes
+        let cstr_args = args
+            .into_iter()
+            .map(CString::new)
+            .collect::<std::result::Result<Vec<_>, _>>()?;
+
+        Self::_new(&cstr_args)
+    }
+
+    fn _new(cstr_args: &[CString]) -> Result<Self> {
+        // Get pointers to the strings
+        let mut c_args = cstr_args
+            .iter()
+            .map(|x| x.as_ptr() as *mut c_char) // These strings don't actually get modified, the C API is just not const-correct
+            .chain(std::iter::once(null_mut())) // Null-terminate the list
+            .collect::<Vec<_>>();
+
+        unsafe {
+            Ok(Self {
+                c_options: gdal_sys::GDALMultiDimTranslateOptionsNew(
+                    c_args.as_mut_ptr(),
+                    null_mut(),
+                ),
+            })
+        }
+    }
+
+    /// Returns the wrapped C pointer
+    ///
+    /// # Safety
+    /// This method returns a raw C pointer
+    ///
+    pub unsafe fn c_options(&self) -> *mut GDALMultiDimTranslateOptions {
+        self.c_options
+    }
+}
+
+impl Drop for MultiDimTranslateOption {
+    fn drop(&mut self) {
+        unsafe {
+            gdal_sys::GDALMultiDimTranslateOptionsFree(self.c_options);
+        }
+    }
+}
+
+impl TryFrom<Vec<&str>> for MultiDimTranslateOption {
+    type Error = GdalError;
+
+    fn try_from(value: Vec<&str>) -> std::result::Result<Self, Self::Error> {
```

Comment:
Does this work?

```suggestion
    fn try_from(value: Vec<&str>) -> Result<Self> {
```

Reply:
[23957df](https://github.com/georust/gdal/pull/289/commits/23957df6195fc5557422ffd81aa724c7aa4aaaf1)